test/integration/Sandwich.t.sol

```diff
@@ -0,0 +1,221 @@
+        MAX_IDLE_LIQUIDITY_PERCENTAGE = 0;
+        super.setUp();
+        deployEverlong();
+        sandwich_short_instant(
+            _shortAmount,
+            _attackerDepositAmount,
+            _bystanderDepositAmount
+        );
+    }
+
+    /// @dev Tests the short sandwich scenario with idle liquidity.
+    function testFuzz_sandwich_short_instant_idle(
+        uint256 _shortAmount,
+        uint256 _attackerDepositAmount,
+        uint256 _bystanderDepositAmount
+    ) external {
+        TARGET_IDLE_LIQUIDITY_PERCENTAGE = 0.1e18;
+        MAX_IDLE_LIQUIDITY_PERCENTAGE = 0.2e18;
+        super.setUp();
+        deployEverlong();
+        sandwich_short_instant(
+            _shortAmount,
+            _attackerDepositAmount,
+            _bystanderDepositAmount
+        );
+    }
+
+    /// @dev Tests the lp sandwich scenario with no idle liquidity.
+    function testFuzz_sandwich_lp_instant_no_idle(
+        uint256 _lpDeposit,
+        uint256 _attackerDepositAmount,
+        uint256 _bystanderDepositAmount
+    ) external {
+        TARGET_IDLE_LIQUIDITY_PERCENTAGE = 0;
+        MAX_IDLE_LIQUIDITY_PERCENTAGE = 0;
+        super.setUp();
+        deployEverlong();
+        sandwich_lp_instant(
+            _lpDeposit,
+            _attackerDepositAmount,
+            _bystanderDepositAmount
+        );
+    }
+
+    /// @dev Tests the lp sandwich scenario with idle liquidity.
+    function testFuzz_sandwich_lp_instant_idle(
+        uint256 _lpDeposit,
+        uint256 _attackerDepositAmount,
+        uint256 _bystanderDepositAmount
+    ) external {
+        TARGET_IDLE_LIQUIDITY_PERCENTAGE = 0.1e18;
+        MAX_IDLE_LIQUIDITY_PERCENTAGE = 0.2e18;
+        super.setUp();
+        deployEverlong();
+        sandwich_lp_instant(
+            _lpDeposit,
+            _attackerDepositAmount,
+            _bystanderDepositAmount
+        );
+    }
+
+    /// @dev Tests the following scenario:
+    ///      1. First an innocent bystander deposits into Everlong. At that time, we
+    ///         also call rebalance to invest Everlong's funds.
+    ///      2. Next, an attacker opens a short. This should decrease Everlong's total
+    ///         assets.
+    ///      3. The attacker then deposits into Everlong. They should receive more
+    ///         shares than the bystander.
+    ///      4. Next, the attacker closes their short. They will lose some money on
+    ///         fees due to this.
+    ///      5. Finally, the attacker withdrawals from Everlong.
+    function sandwich_short_instant(
+        uint256 _shortAmount,
+        uint256 _attackerDepositAmount,
+        uint256 _bystanderDepositAmount
+    ) public {
+        // Alice is the attacker, and Bob is the bystander.
+        address attacker = alice;
+        address bystander = bob;
+
+        // The bystander deposits into Everlong.
+        _bystanderDepositAmount = bound(
+            _bystanderDepositAmount,
+            1e18,
```

Comment:
For this minimum and all of the other minimums, it would be better to use the minimum transaction amount. Maybe I'm missing something for this deposit amount, but it would definitely be better for `_shortAmount`.

Reply:
Agreed. I decreased to `minTxAmount` and was able to fix some issues but not all.

Decreased to `5 * MINIMUM_TRANSACTION_AMOUNT` and put in a TODO to lower it further.

I'm thinking that the updates that go into place for not rebalancing on deposit will likely fix this, which is why I'm kinda punting.